cardinal/server/server_test.go

```diff
@@ -122,8 +117,51 @@ func setTestTimeout(t *testing.T, timeout time.Duration) {
 	}()
 }
 
+func TestHealthEndpoint(t *testing.T) {
+	//setTestTimeout(t, 10*time.Second)
+	w := inmem.NewECSWorldForTest(t)
+	sendTx := ecs.NewTransactionType[SendEnergyTx, SendEnergyTxResult]("sendTx")
+	assert.NilError(t, w.RegisterTransactions(sendTx))
+	w.AddSystem(func(world *ecs.World, queue *transaction.TxQueue, _ *log.Logger) error {
+		return nil
+	})
+	assert.NilError(t, w.LoadGameState())
+	txh := makeTestTransactionHandler(t, w, DisableSignatureVerification())
+	resp, err := http.Get("http://localhost:4040/health")
+	assert.NilError(t, err)
+	assert.Equal(t, resp.StatusCode, 200)
+	body := make(map[string]bool)
+	err = json.NewDecoder(resp.Body).Decode(&body)
+	assert.NilError(t, err)
+	v, ok := body["is_server_running"]
+	assert.Assert(t, ok)
+	assert.Assert(t, v)
+	v, ok = body["is_game_loop_running"]
+	assert.Assert(t, ok)
+	assert.Assert(t, !v) //game loop is not running!
+	ctx := context.Background()
+	w.StartGameLoop(ctx, time.Tick(1*time.Second), nil)
+	isGameLoopRunning := false
+	for !isGameLoopRunning {
+		time.Sleep(200 * time.Millisecond)
+		resp, err = http.Get("http://localhost:4040/health")
+		assert.NilError(t, err)
+		assert.Equal(t, resp.StatusCode, 200)
+		err = json.NewDecoder(resp.Body).Decode(&body)
+		v, ok = body["is_server_running"]
+		assert.Assert(t, ok)
+		assert.Assert(t, v)
+		isGameLoopRunning, ok = body["is_game_loop_running"]
+		assert.Assert(t, ok)
+	}
+	gameObject := NewGameManager(w, txh.Handler)
+	err = gameObject.Shutdown()
+	assert.NilError(t, err)
```

Comment:
do we need this stuff here if we're not testing the responses after shutdown?

Reply:
Doesn't hurt to have this. It's proper shutdown anyway. But yes there is a reason why I put this here. 

I put this here because I think there's a slight bug in our test code. There's a lingering side effect that occurs in our code. 

It may not be reproducible on your computer but what I see happening is that if I remove this shutdown code another test will begin failing at the storage layer. You might be able to repo and see this error: `{"level":"panic","error":"dial tcp 127.0.0.1:50596: connect: connection refused","time":"2023-09-25T21:03:57-07:00","message":"Error running Tick in Game Loop."}`. miniredis is basically doing some weird shit here as that port changes. 

If I run the tests individually it works, if I run them together we have a problem....

What I think is going on is we aren't properly shutting down something related to miniredis. The real redis likely handles all of this better so this is a subtle error related to tests and outside of the scope of this ticket. 

I have this on my mind. Likely will make another issue for it soon, but it's not related to what I'm doing here. Calling the shutdown code temporarily bypasses the error on my end.

Reply:
ah i see. we might benefit from using that [TestSuite](https://pkg.go.dev/github.com/stretchr/testify/suite) thingy. that way we can define pre-post test run functions to handle these things, and don't have to remember to do it each test. but we can just open another issue for this as you said its OOS here

Reply:
The fix is likely exposing the redis Shutdown method.

Reply:
I suspect the problem is 2 instances of miniredis are trying to bind to the same port. I agree that exposing a redis shutdown is the right thing (eventually, not necessarily in this PR)